### PR TITLE
fix: KafkaBoundedSupervisorTest  flaky test

### LINKED ISSUE #19542: Flaky Test: KafkaBoundedSupervisorTest
```

Run mikepenz/action-junit-report@v6
📘 Reading input values
📦 Process test results
  Preparing 1 report as configured.
  embedded-tests/target/test-classes/org/apache/druid/testing/embedded/indexing/KafkaBoundedSupervisorTest.class:91 | Timed out waiting for event after [60,000]ms 180.259s
  ℹ️ Posting with conclusion 'failure' to https://github.com/apache/druid/pull/19266 (sha: e9b3b82bb210332e5ca08792856341b80257206d)
🚀 Publish results

```

Ref https://github.com/apache/druid/pull/19266

### PR DESCRIPTION
**Pull request by Shekharrajak**
Fixes https://github.com/apache/druid/issues/19542 .

### Description
embedded-tests: allow per-call timeout for ingestion wait, bump KafkaBoundedSupervisorTest to 120s

The bounded
supervisor cold-start pipeline (post supervisor -> schedule task ->
assign partitions -> consume from Kafka -> publish -> COMPLETED) can
exceed the cluster default 60s on a cold CI runner.

#### Release note

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [x] a release note entry in the PR description.
- [x] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/StreamIndexTestBase.java [hunk 1]
```diff
@@ -131,15 +131,34 @@ protected KinesisSupervisorSpec createKinesisSupervisor(KinesisResource kinesis,
...
+  }
+
+  /**
+   * Same as {@link #waitUntilPublishedRecordsAreIngested(int)} but with an explicit timeout in millis.
+   * Use for ingestion paths with a heavier task lifecycle (e.g. bounded supervisor cold start) where the
+   * cluster default may not allow enough headroom on CI.
+   */
+  protected void waitUntilPublishedRecordsAreIngested(int expectedRowCount, Long timeoutMillis)
+  {
+    if (timeoutMillis == null) {
```
Comment (kfaraz) [inline embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/StreamIndexTestBase.java]: Instead of making the timeout nullable in this method and then adding an if-else anyway, do the following:
- Leave the original method unchanged.
- Add a new method which accepts a primitive `long timeout`, which must always be non-null and greater than 0.

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 3 of 3 changed files.

---

This is an automated review by Codex GPT-5.5

File: embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/StreamIndexTestBase.java (hunk 1 above)
Comment (Shekharrajak) [reply embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/StreamIndexTestBase.java]: Yes the above method is the original one https://github.com/apache/druid/pull/19543/changes#diff-cf5b6e87308f6b79c993fbd8d032d1e980d74083d8397267ba646e8921679379R136 

and `protected void waitUntilPublishedRecordsAreIngested(int expectedRowCount, Long timeoutMillis)` is method overloading.

**Comment by Shekharrajak** [comment]
@jtuglu1 , please have a look - this flaky tests occurring here https://github.com/apache/druid/pull/19266

File: embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/StreamIndexTestBase.java (hunk 1 above)
Comment (jtuglu1) [reply embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/StreamIndexTestBase.java]: Let's mark the `timeoutMillis` as @Nullable and then I'm fine with this change unless @kfaraz has objections

**Review by jtuglu1 — APPROVED** [review]

File: embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/StreamIndexTestBase.java (hunk 1 above)
Comment (kfaraz) [reply embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/StreamIndexTestBase.java]: @Shekharrajak , you shouldn't need to have an if-else at all since you already have 2 separate methods.
Old method should have the code block inside the `if` and the new method should have the code block inside the `else`. Old method does not need to call the new method since we are not commoning out any code, both method bodies are mutually exclusive.

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 3 of 3 changed files.

---

This is an automated review by Codex GPT-5.5

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 3 of 3 changed files.

---

This is an automated review by Codex GPT-5.5

**Review by kfaraz — APPROVED** [review]
